### PR TITLE
refactor combine destinationrule

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

File: pilot/pkg/model/push_context_test.go [hunk 1]
```diff
@@ -860,11 +860,11 @@ func scopeToSidecar(scope *SidecarScope) string {
 func TestSetDestinationRule(t *testing.T) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/push_context_test.go]: This ut was not right

File: pilot/pkg/model/destination_rule.go [hunk 2]
```diff
@@ -18,61 +18,49 @@ import (
...
 	networking "istio.io/api/networking/v1alpha3"
-
-	"istio.io/istio/pkg/config/host"
 )
 
 // This function merges one or more destination rules for a given host string
 // into a single destination rule. Note that it does not perform inheritance style merging.
 // IOW, given three dest rules (*.foo.com, *.foo.com, *.com), calling this function for
 // each config will result in a final dest rule set (*.foo.com, and *.com).
 func (ps *PushContext) combineSingleDestinationRule(
```
Comment (ramaraochavali) [inline pilot/pkg/model/destination_rule.go]: nit: may be we should call this `mergeDestinationRule`?

File: pilot/pkg/model/destination_rule.go [hunk 3]
```diff
@@ -18,61 +18,49 @@ import (
...
-		config:  &copyDestRuleConfig,
-	}
-	for _, subset := range rule.Subsets {
-		combinedDestRuleMap[resolvedHost].subsets[subset.Name] = struct{}{}
+		return
 	}
-	combinedDestRuleHosts = append(combinedDestRuleHosts, resolvedHost)
 
-	return combinedDestRuleHosts
+	p.hosts = append(p.hosts, resolvedHost)
```
Comment (ramaraochavali) [inline pilot/pkg/model/destination_rule.go]: Can you add a small comment here that DestinationRule does not exist for the resolved host so adding it?

File: pilot/pkg/model/destination_rule.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/model/destination_rule.go]: I am ok

**Pull request merged**